### ISSUE TITLE
Add proper touch support (for Wayland only)

### ISSUE THREAD

**Issue by Leleat** [Leleat/Tiling-Assist#63]
If I am understanding https://gitlab.gnome.org/GNOME/mutter/-/issues/295 correctly, there is no proper way for me support touch on a Xorg session. **On Wayland the only thing missing should be the ability to open the window preview of the tiling popup.**

However, I don't know how touch works and I also don't have a touch device, so I cannot implement this. Someone else needs to do this.

**Comment by cameronaw13** [comment]
From another project that I'm referencing ([gjs-osk](https://github.com/Vishram1123/gjs-osk)), it seems that they use Clutter's events and vfunctions to detect pointer, touch, and event pen events. For pointers and pens, `BUTTON_PRESS`, `MOTION`, and `BUTTON_RELEASE` are used. For touch, `TOUCH_BEGIN`, `TOUCH_UPDATE`, and `TOUCH_END` are used. For all 3 event types they can then use `event.get_coords()` to find the coordinates for any input events. Not sure how to implement this within the current moveHandler.js yet as I have no experience with gjs but I am able to test it using a convertible laptop.

**Comment by cameronaw13** [comment]
I've been looking around for others' solutions and found that [Fly-Pie already created an implementation of what I proposed above](https://github.com/Schneegans/Fly-Pie/blob/main/src/extension/Menu.js). It seems that they use a background modal that captures Clutter input events which is then sent to the menu where some checks are performed before ultimately finding the input's coordinates.

I feel like something of this nature could be implemented within this project if it's able to mesh well with the moveHandler class and Meta.

**Comment by Leleat** [comment]
Thanks for the work, I'll look into the other projects. I've wanted to rework the 'moveHandler' at some point. I'll try to get the touch support into it then as well.

**Comment by krzemyczek** [comment]
Oh, old issue, I found this project works the best and touch support would make it even better :)

**Comment by Leleat** [comment]
> Oh, old issue, I found this project works the best and touch support would make it even better :)

I've recently bought the new Lenovo Yoga Slim 7i, which has a touchscreen. So I can test the touch support now myself and I plan to add it to Tiling Assistant.

The main problem ATM is that Linux doesn't run properly on the Yoga yet (without building some code from source). So I am waiting for Linux to work out of the box to install it on the Laptop.

**Comment by krzemyczek** [comment]
> Lenovo Yoga Slim 7i,

Yeah, I looked at this series too, a shame, because I felt in love with them.
 But good to hear it :) 

Thank you BTW for your extension
